main/lsp/requests/rename.cc

```diff
@@ -44,12 +44,24 @@ bool isSubclass(const core::GlobalState &gs, core::SymbolRef root, core::SymbolR
     }
     visited[s.classOrModuleIndex()] = true;
 
-    auto super = s.data(gs)->superClass();
-    if (super.exists()) {
-        memoized[s.classOrModuleIndex()] = isSubclass(gs, root, super, memoized, visited);
+    if (s.data(gs)->derivesFrom(gs, root)) {
```

Comment:
`derivesFrom` actually checks the superclass and mixins in the proper order.

I think we can actually consolidate this if/else into one loop so we can avoid doing the expensive `derivesFrom` check on every symbol and instead make use of our memoization.

Reply:
e.g., iterate thru all mixins until one `isSubclass` then check parent class